### PR TITLE
Add routing for news detail

### PR DESCRIPTION
**Pull request by annanaj**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,22 +1,55 @@
...
+		document.body.addEventListener('click', handleLinkClick);
+		return () => {
+			document.body.removeEventListener('click', handleLinkClick);
+		};
+	}, [navigate]);
+
 	return (
-		<Router>
+		<>
 			<Navbar />
```
Comment (crishpeen) [inline src/App.tsx:38]: can you make the logo and page name redirect to the hp?

File: src/App.tsx [hunk 2]
```diff
@@ -1,22 +1,55 @@
...
-			<Routes>
+			<Routes location={location}>
 				<Route path="/" element={<NewsList />} />
-				<Route path={routes.themesList} element={<ThemesList />} />
 				<Route path={routes.newsList} element={<NewsList />} />
```
Comment (crishpeen) [inline src/App.tsx:41]: seo would probably punish you, as you have the same content on the URLs (/, newsList)

File: package.json [hunk 3]
```diff
@@ -22,6 +22,7 @@
 		"@ntvr/eslint-config": "^1.0.2",
 		"@ntvr/stylelint-config": "^3.0.0",
 		"@stylistic/eslint-plugin": "^2.0.0",
+		"@tailwindcss/postcss7-compat": "npm:@tailwindcss/postcss7-compat@2.0.1",
```
Comment (crishpeen) [inline package.json]: couldn't this be just 
```suggestion
		"@tailwindcss/postcss7-compat": "2.0.1",
```
?

File: src/App.tsx [hunk 4]
```diff
@@ -1,22 +1,55 @@
...
 import PageNotFound from './pages/PageNotFound/PageNotFound';
 import ThemesList from './pages/ThemesList/ThemesList';
 
-export default function App() {
+function AppContent() {
+	const location = useLocation();
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		const handleLinkClick = (event: any) => {
```
Comment (crishpeen) [inline src/App.tsx]: any :(

File: src/hooks/useFetchNewsData.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+	if (newsData && newsData.length > 0) {
+		data = newsData;
+	} else if (cachedData) {
+		data = JSON.parse(cachedData);
+	} else {
+		data = [];
+	}
+
+	return { data, loading };
```
Comment (crishpeen) [inline src/hooks/useFetchNewsData.ts]: it would be nice to set what types the returned values are.

File: src/pages/NewsItem/NewsItem.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import placeholder from '../../assets/placeholder.svg';
```
Comment (crishpeen) [inline src/pages/NewsItem/NewsItem.tsx:1]: no react import or anything? :O

File: src/pages/NewsItem/NewsItem.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+	const { data, loading } = useFetchNewsData();
+	const { id } = useURL();
+	const newsItem = data.find((item: NewsData) => item.time_published === id);
+
+	return (
+		<div className="container mt-24 mx-auto px-7">
+			<div className="flex items-center gap-12 mb-7">
+				{loading ? (
+					<div>
+						<p className="mt-2">Loading article data...</p>
```
Comment (crishpeen) [inline src/pages/NewsItem/NewsItem.tsx]:
```suggestion
						<p className="mt-2">Loading article data…</p>
```
:D

File: src/pages/NewsList/NewsList.tsx [hunk 8]
```diff
@@ -128,20 +70,21 @@ export default function NewsList() {
 
 					{filteredData && filteredData.length > 0 ? (
 						<div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-5 min-w-96">
-							{data.map((newsItem: NewsData) => (
-								<div
+							{filteredData.map((newsItem: NewsData) => (
+								<a
```
Comment (crishpeen) [inline src/pages/NewsList/NewsList.tsx]: In another app I would like to see this. This should be <article> and the link should be done elseway. Either just wrap the image and title, etc. Or add one link somewhere with correct content (a11y reader would probably try to read everything in this element as a link) and with pseudo element covering the whole article.

File: src/pages/NewsList/NewsList.tsx [hunk 9]
```diff
@@ -128,20 +70,21 @@ export default function NewsList() {
...
 									key={`${newsItem.url}-${newsItem.time_published}`}
-									className="flex flex-col h-full p-6 mb-6 bg-gray-50 rounded-lg shadow-lg"
+									href={`${routes.newsItem}?id=${newsItem.time_published}`}
```
Comment (crishpeen) [inline src/pages/NewsList/NewsList.tsx]: in some bigger app, you would probably have a function for this, as you will want to keep the URL structure in one place.

File: package.json (hunk 3 above)
Comment (annanaj) [reply package.json]: i will rollback this, as it was connected with tailwind autocomplete, with no luck

File: src/App.tsx (hunk 2 above)
Comment (annanaj) [reply src/App.tsx:41]: adding homepage and some content

File: package.json [hunk 10]
```diff
@@ -16,6 +16,7 @@
 		"react-dom": "^18.2.0",
 		"react-router-dom": "^6.22.3",
 		"stylelint-order": "^6.0.4",
+		"tailwind": "4",
```
Comment (crishpeen) [inline package.json]: https://github.com/thenativeweb/tailwind/issues/19

File: src/pages/Homepage/Homepage.tsx [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+					core stock API, fundamental data, and technical indicator APIs,
+					can provide you with a 360-degree view of the financial market and
+					the broader economy.
+				</p>
+			</div>
+			<h2 className="mb-4 text-2xl font-medium">Latest updates</h2>
+			{data && data.length > 0 ? (
+				<div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-5 min-w-96 mb-10">
+					{data.slice(0, 3).map((newsItem: NewsData) => (
+						<div
```
Comment (crishpeen) [inline src/pages/Homepage/Homepage.tsx]:
```suggestion
						<article
```
?

File: src/pages/NewsList/NewsList.tsx [hunk 12]
```diff
@@ -128,36 +71,38 @@ export default function NewsList() {
 
 					{filteredData && filteredData.length > 0 ? (
 						<div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-5 min-w-96">
-							{data.map((newsItem: NewsData) => (
+							{filteredData.map((newsItem: NewsData) => (
 								<div
```
Comment (crishpeen) [inline src/pages/NewsList/NewsList.tsx]: <article?

**Pull request merged**